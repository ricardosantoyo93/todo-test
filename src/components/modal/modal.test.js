import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ShallowRenderer from 'react-test-renderer/shallow';

import { Modal } from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Modal', () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(<Modal />);
  });
  
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Modal />);
    expect(tree).toMatchSnapshot();
  });

  test('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });
});
