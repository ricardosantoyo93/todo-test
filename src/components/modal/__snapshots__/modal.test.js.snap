// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<Modal
  actions={
    Array [
      <Button
        flat={true}
        modal="close"
        node="button"
        waves="green"
      >
        Close
      </Button>,
    ]
  }
  bottomSheet={false}
  fixedFooter={false}
  header="Add new card"
  id="modal-card-undefined"
  open={false}
  options={
    Object {
      "dismissible": true,
      "endingTop": "10%",
      "inDuration": 250,
      "onCloseEnd": null,
      "onCloseStart": null,
      "onOpenEnd": null,
      "onOpenStart": null,
      "opacity": 0.5,
      "outDuration": 250,
      "preventScrolling": true,
      "startingTop": "4%",
    }
  }
  root={<body />}
  trigger={
    <Button
      floating={true}
      icon={
        <Icon>
          add
        </Icon>
      }
      id="modal-open-undefined"
      node="button"
      small={true}
      waves="light"
    />
  }
>
  <div>
    <ForwardRef
      id="modal-title-undefined"
      label="Title"
      onChange={[Function]}
    />
    <Textarea
      id="modal-textarea-undefined"
      label="Description"
      onChange={[Function]}
    />
    <Button
      id="modal-submit"
      node="button"
      onClick={[Function]}
      type="button"
    >
      Submit
    </Button>
  </div>
</Modal>
`;
